xblock/mixins.py

```diff
@@ -507,32 +507,44 @@ def xml_text_content(self):
             return None
 
     @classmethod
-    def _set_field_if_present(cls, block, name, value):
+    def _set_field_if_present(cls, block, name, value, attrs):
         """Sets the field block.name, if block have such a field."""
         if name in block.fields:
             value = (block.fields[name]).from_string(value)
-            setattr(block, name, value)
+            if "none" in attrs and attrs["none"] == "true":
+                setattr(block, name, None)
+            else:
+                setattr(block, name, value)
         else:
             logging.warn("XBlock %s does not contain field %s", type(block), name)
 
     def _add_field(self, node, field_name, field):
-        """Add xml representation of field to node.
+        """
+        Add xml representation of field to node.
 
         Depending on settings, it either stores the value of field
         as an xml attribute or creates a separate child node.
         """
         value = field.to_string(field.read_from(self))
+        text_value = "" if value is None else value
 
-        if value is None:
-            value = ""
+        # Is the field type supposed to serialize the fact that the value is None to XML?
+        none_to_xml_attr = field.none_to_xml and value is None
```

Comment:
How about `save_none_as_xml_attr`? Short for "Save a None value as an XML attribute on an XML element."